src/lib/sidenav/sidenav.md

```diff
@@ -82,3 +82,17 @@ is clicked or <kbd>ESCAPE</kbd> is pressed. To add custom logic on backdrop clic
   <mat-sidenav disableClose (keydown)="customKeydownHandler($event)"></mat-sidenav>
 </mat-sidenav-container>
 ```
+
+### Resizing an open sidenav
+By default, Material will only measure and resize the drawer container in a few key moments
+(on open, on window resize, on mode change) in order to avoid layout thrashing, however there
+are cases where this can be problematic. If your app requires for a drawer to change its width
+while it is open, you can use the `autosize` option to tell Material to continue measuring it.
+Note that you should use this option **at your own risk**, because it could cause performance
+issues.
+
+```html
+<mat-sidenav-container [autosize]="true">
```

Comment:
can we make this an embedded example? (https://github.com/angular/material2/pull/7775 will make the rest of these embedded examples as well).

Maybe also add an example to the demo app so we can manually test

Reply:
I'm not sure what an embedded example would look like in this case, aside from a sidenav that resizes in a `setTimeout`, but that's not very representative of a real use case.

Reply:
You could have an icon drawer that expands to show full labels when the user hovers or clicks a button.

Or just have a button that toggles itself between shorter and longer width when clicked, not a practical use case but demonstrates how to use the feature and lets us manually test it

Reply:
Alright, I've added a live example.